go/vt/vtgate/tabletgateway.go

```diff
@@ -301,10 +301,16 @@ func (gw *TabletGateway) withRetry(ctx context.Context, target *querypb.Target,
 					err = vterrors.Errorf(vtrpcpb.Code_CLUSTER_EVENT, "current keyspace is being resharded")
 					continue
 				}
-				if kev.PrimaryIsNotServing(target) {
```

Comment:
not neccessary to fix in this PR, but damn, this method needs to be cleaned up. deep indentation and long method makes it difficult to follow